### PR TITLE
Add SUSE Manager settings dialog component

### PR DESCRIPTION
**Pull request by dottorblaster**
# Description
Adding SUSE Manager settings dialog component. Examples on Storybook.

## How was this tested?
Storybook stories, Jest tests added

### CODE REVIEW

**Comment by dottorblaster** [comment]
Dependsa on #2329 but still reviewable

File: assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            placeholder="Starts with -----BEGIN CERTIFICATE-----"
+            error={hasError('ca_cert', errors)}
+            onChange={({ target: { value } }) => {
+              setCertificate(value);
+              onClearErrors();
+            }}
+          />
+        ) : (
+          <div className="col-span-4 flex flex-row items-center justify-start p-5 border border-gray-200 rounded-md">
+            <EOS_LOCK_OUTLINED className="mr-3" size="25" />
```
Comment (nelsonkopliku) [inline assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx:70]: I guess we could/should reuse `CertificateUploadDate` component here. Or an adjustment of it to make it reusable here.

File: assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        </p>
+      </div>
+      <div className="flex flex-row w-80 space-x-2">
+        <Button
+          disabled={loading}
+          onClick={() => {
+            const payload = {
+              url,
+              username,
+              ...(editingCertificate && { certificate }),
```
Comment (nelsonkopliku) [inline assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx:135]: Let's just keep in mind that the api expects `ca_cert` in the payload: this ends up as `certificate`.

If we intend to have an extra transformation layer before sending to the API then ok.

File: assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    errors: {
+      description: 'OpenAPI errors coming from backend validation',
+    },
+  },
+};
+
+export const Default = {
+  args: {
+    open: false,
```
Comment (nelsonkopliku) [inline assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.stories.jsx:45]: What about a trigger in the stories to open the dialog? Or alternatively `open: true` to all of them, if it does not become too invasive.

**Review by nelsonkopliku — CHANGES_REQUESTED** [review]
Looks good! 
The main change needed imho is adding the ability in stories to actually open the dialog.

The others can be deferred in following PRs, if you wish.

**Comment by dottorblaster** [comment]
@nelsonkopliku Thanks for the review! I was thinking the same about the button, initially I provided `open: true` to the stories but in the documentation page it became really cumbersome with those being modals.

Then I thought we could just use the controls plugin

immagine

What do you think?

**Comment by nelsonkopliku** [comment]
@dottorblaster you're completely right, makes totally sense. It didn't immediately pop to my mind :smile:

**Comment by dottorblaster** [comment]
@nelsonkopliku no problemo :+1: onto the other stuff you spotted :dash:

**Review by nelsonkopliku — APPROVED** [review]
Alright, good with storybook controls :rocket: 

For the other things up to you about when to tackle :+1:

**Review by jamie-suse — APPROVED** [review]

File: assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+import { EOS_LOCK_OUTLINED } from 'eos-icons-react';
+
+import Button from '@common/Button';
+import Modal from '@common/Modal';
+import Input, { Password, Textarea } from '@common/Input';
+import Label from '@common/Label';
+
+import { hasError } from './errors';
+
+const defaultErrors = [];
```
Comment (jamie-suse) [inline assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx:13]: Does this have a purpose?

Comment (dottorblaster) [reply assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx:13]: It does! If you specify empty arrays, empty objects or functions (mainly the first two) as default props without declaring them as `const` you incur into their reference constantly changing when rendering and components just get a new prop ref every time they render, ending up in endless rerendering. Your browser gets easily stuck on those, crashing and looping.

Comment (dottorblaster) [reply assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx:13]: More info here https://github.com/facebook/react/issues/18123

Comment (jamie-suse) [reply assets/js/common/SuseManagerSettingsDialog/SuseManagerSettingsModal.jsx:13]: ahhh very tricky! Good to know :brain:

**Pull request merged**